server/src/testFixtures/java/io/crate/testing/QueryTester.java

```diff
@@ -86,12 +87,9 @@ public Builder(Path tempDir,
                        Version indexVersion,
                        String createTableStmt,
                        AbstractModule... additionalModules) throws IOException {
-            // Disable OID generation for columns/references in order to be able to compare the query outcome with
-            // expected ones.
-            LongSupplier columnOidSupplier = () -> COLUMN_OID_UNASSIGNED;
             var sqlExecutor = SQLExecutor
                 .builder(clusterService, additionalModules)
-                .setColumnOidSupplier(columnOidSupplier)
+                .setColumnOidSupplier(new Metadata.ColumnOidSupplier(COLUMN_OID_UNASSIGNED))
```

Comment:
The whole intention of not using a real OID supplier was that we can still assert in all query tests for real column names. Using OID's will decrease readability completely, it would be very hard to see if the expected outcomes are valid. What about @mfussenegger suggestion to add test variants for the concrete issues which do not assert the query output? I highly suggest to move into such a direction instead.

Reply:
Also, I made the `ColumnOidSupplier` private by intent, to not use it in such tests, but instead, if really needed, use a custom implementation (its `LongSupplier` so a custom impl is very simple).

Reply:
Sorry for pushing in the wrong direction (I meant using custom `LongSupplier` impl but asserting oids everywhere is indeed not readable)

Basically https://github.com/crate/crate/commit/99e697e00e84bfc8a34ad1a501bbc686dc07dc91 was okay in that regard that it was a dedicated test, with simple custom supplier to kick new logic the only adjustment would be to follow 
> Similar to test_is_null_on_columns_without_doc_values (minus the _field_names stuff)
and assert not `queryTester.toQuery` output but `queryTester.runQuery` output -> which would be failing without the fix even with custom "return constant" supplier.

Reply:
Thank you both for looking into this. Could I make just one more [attempt](https://github.com/crate/crate/pull/14908/commits/f819a98e584092010831ac4d95740bb6550effb2) hoping the existing tests account for oids.

Reply:
I have 2 questions
1.  > and actually found another missing conversion from NumTermsPerDocQuery.

Is this covered somewhere in the test? I see change in relevant `testNeqAnyOnArrayLiteral()` but test_is_null_on_columns_without_doc_values seems to be irrelevant - I feel like problems in numTermsPerDoc should be somewhere in `ArrayLengthQueryBuilderTest`

2. Current approach with checking query and keep it human readable via `queryTester.columnNameToOid` looks fine to me but let's wait for other opinions (cc @seut, @mfussenegger). I think another approach is to not assert query at all but instead check output of `queryTester.run()`?

Reply:
https://github.com/crate/crate/pull/14908/commits/f819a98e584092010831ac4d95740bb6550effb2 allows lucene query tests to use oids directly or `queryTester.columnNameToOid("x")` to make it more readable.

1. The changes to `test_is_null_on_columns_without_doc_values` was just to describe how existing query tests will be updated.
2. > I think another approach is to not assert query at all but instead check output of queryTester.run()

I think `runQuery` is slightly more time consuming since we need to index and execute, we could also wait for others opinions on this.
3. https://github.com/crate/crate/commit/3acc06c5973f795328f450e3455fe1d0f7b783f5 will be updated to use `queryTester.columnNameToOid`, so I am hoping all existing query tests will be updated accordingly.